### ISSUE TITLE
incorrectly extracts files (leaves behind ADF blocks)

### ISSUE THREAD

**Issue by jmtd** [mikeakohn/amiga_recovery#1]
I've been using `amiga_recovery` to extract files from ADF disk images (created by XCOPY on a real Amiga, dumping to a Gotek virtual floppy disk adapter). I'm pointing `amiga_recovery` at each disk image and then running `copy all <dest dir>`

When trying to read some of the Deluxe Paint IV-authored IFF/ILBM files that resulted I was hitting a lot of problems. There's 24 bytes prior to the magic that denotes the start of these files ("FORM") e.g.

```
00000000  00 00 00 08 00 00 03 d4  00 00 00 01 00 00 01 e8  |................|
00000010  00 00 03 e2 c4 b2 3d e4  46 4f 52 4d 00 00 10 be  |......=.FORM....|
```

A friend helping me (@sladen) with decoding realised that these are "Amiga sector headers, matching the 'blkhdr' structure in  http://www.weihenstephan.org/~michaste/pagetable/amiga/extract-adf.c"

**Comment by mikeakohn** [comment]
Weird.  Is it possible you can email me a disk image with a file that doesn't work so I can debug it?

**Comment by jmtd** [comment]
Sure, no problem.

**Comment by sladen** [comment]
Turns out the 24-byte sector headers are present every 512 bytes:

```
$ hexdump -C Ice | grep '00 00 00 08' | head -n4
00000000  00 00 00 08 00 00 03 75  00 00 00 01 00 00 01 e8  |.......u........|
00000200  00 00 00 08 00 00 03 75  00 00 00 02 00 00 01 e8  |.......u........|
00000400  00 00 00 08 00 00 03 75  00 00 00 03 00 00 01 e8  |.......u........|
00000600  00 00 00 08 00 00 03 75  00 00 00 04 00 00 01 e8  |.......u........|
```

What should be happening is that the sector header should be dropped, and only the remaining 488 bytes in the sector copied through into the file as data.  The overall file length *is* correct.

**Comment by mikeakohn** [comment]
That's really odd.  I wrote this utility so I could extract data from my old Amiga 4000 harddisks and never had a problem with it.

**Comment by mikeakohn** [comment]
I took a quick look into the .adf file you sent me last night. I extracted a .info file from your disk and one from one of my images that I used before.  My .info file started with the correct magic number of 0xe310 while yours started with (as you stated) 24 bytes and then the magic number 0xe310.  One visible difference in what you’re doing and what I’m doing, I wrote this program to read in harddisks, and not floppies.  So on startup it tries to find partition tables and such (it looks for a magic number of RDSK.

So amiga_recovery does a search and does find the start of a partition (magic number of DOS\0 on yours and DOS\1 on mine).  Seems that DOS\0 is an OFS file system while DOS\1 is AFS and if I’m understanding this right, the OFS file system does keep an extra 24 bytes at the start of each block.  I can detect OFS and remove the 24 bytes and I think this will take care of it.
